src/__experimental__/components/date/date.component.js

```diff
@@ -13,45 +13,29 @@ import withUniqueName from '../../../utils/helpers/with-unique-name';
 const defaultDateFormat = 'DD/MM/YYYY';
 
 class BaseDateInput extends React.Component {
-  static propTypes = {
-    ...Textbox.propTypes,
-    /** Automatically focus on component mount */
-    autoFocus: PropTypes.bool,
-    /** Used to provide additional validations on composed components */
-    internalValidations: PropTypes.array,
-    /** Minimum possible date YYYY-MM-DD */
-    minDate: PropTypes.string,
-    /** Maximum possible date YYYY-MM-DD */
-    maxDate: PropTypes.string,
-    /** Specify a callback triggered on blur */
-    onBlur: PropTypes.func,
-    /** Specify a callback triggered on change */
-    onChange: PropTypes.func,
-    /** Specify a callback triggered on focus */
-    onFocus: PropTypes.func,
-    /** Name of the input */
-    name: PropTypes.string,
-    /** The current date YYYY-MM-DD */
-    value: PropTypes.string,
-    /** Triggers textbox validation when it's boolean value changes */
-    forceUpdateTriggerToggle: PropTypes.bool
-  };
-
-  static defaultProps = {
-    value: DateHelper.todayFormatted(),
-    internalValidations: [new DateValidator()]
-  };
-
   isBlurBlocked = false;
 
   isOpening = false;
 
+  inputHasFocus = this.props.autoFocus || false;
+
+  isControlled = this.props.value !== undefined;
+
+  adjustedValue = DateHelper.isValidDate(this.props.value) ? this.props.value : DateHelper.todayFormatted();
+
+  initialVisibleValue = this.isControlled ? this.adjustedValue : this.props.defaultValue;
+
   state = {
     isDatePickerOpen: false,
     /** Date object to pass to the DatePicker */
-    selectedDate: DateHelper.stringToDate(this.props.value),
+    selectedDate: DateHelper.stringToDate(DateHelper.formatValue(this.initialVisibleValue)),
     /** Displayed value, format dependent on a region */
-    visibleValue: formatDateToCurrentLocale(this.props.value)
+    visibleValue: formatDateToCurrentLocale(this.initialVisibleValue),
+    /** Stores last valid values to be emitted onBlur if current input is invalid */
+    lastValidEventValues: {
+      formattedValue: formatDateToCurrentLocale(this.initialVisibleValue),
+      rawValue: DateHelper.formatValue(this.initialVisibleValue)
```

Comment:
There's some discussion about whether nested objects in `state` are a good idea :point_right: https://stackoverflow.com/a/51136076

I think it's OK here, but it's worth bearing this in mind :point_right: [Speed up your React Components with flattened states and `React.PureComponent`](https://www.codementor.io/manasjayanth/speed-up-your-react-components-with-flattened-states-and-react-purecomponent-m3hv6mlrf)

Reply:
Yeah agree, in this scenario I think it's fine as it's essentially just maintaining a snapshot of the previous valid values

Reply:
:white_check_mark: